### PR TITLE
feat:  k8s stateless example

### PR DESCRIPTION
**Pull request by bwanglzu**
- [x] Add docker file and k8s configs for pod-1 (encoder).

- [x] Patch health check for the pod.

- [x] Document the steps.

- [ ] Check consistency and tolerance w/w.o. replicas

- [ ] Repeat the previous steps to a `crafter`, `classifier`, `evaluator` and `ranker`.

### CODE REVIEW

File: k8s_example/mwu-encoder/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+**STEP 3**: Deploy to k8s pod:
+
+```
+kubectl apply -f k8s/configmap.yaml 
+kubectl apply -f k8s/deployment.yaml 
+```
+
+It should be noted that we provided a `deployment.yaml` and a `configmap.yaml` in `k8s` folder. `deployment.yaml` specifiy the horizontal scaling capbility with the `replicas` field. Besids, it controls the resource usage and environment variables for the deployment. When applying this deployment, the original entrypoint will be overwritten by the env variables defined in `command`, `args` field and `configmap.yaml`.
```
Comment (JoanFM) [inline k8s_example/mwu-encoder/README.md:30]: How exactly these replicas behave? Specially for index? do they get the same data all the time?

Comment (rutujasurve94) [reply k8s_example/mwu-encoder/README.md:30]: Afaik the compute will be distributed among the replicas so they won't have the exact same data. This is to make the deployment highly available

Comment (JoanFM) [reply k8s_example/mwu-encoder/README.md:30]: so no relation with replicas in terms of replica of data?

Comment (nan-wang) [reply k8s_example/mwu-encoder/README.md:30]: hi, this step is more about how to spin up the pods with k8s. We stick to `replicas=1` for the time being.

**Pull request closed without merging**